T2430 popcorn landing page

stenington opened the pull request:
This is the current state of the popcorn landing page. Brett says you guys will take it from here!

Everything lives under `public`. The page is static html, and the css is straight css (not LESS). It's not as clean as it could be, but my understanding was that this was a stopgap sort of thing.

stenington committed:
[t2430] Replacing 8-bit popcorn landing page with Thimble landing page,
to be turned into popcorn landing page. Added a few nominal differences
to get the transition to popcorn content rolling.

stenington committed:
[t2430] Changed main text content, and removed unnecessary nav.

stenington committed:
[t2430] Bring in video.js and popcorn video

stenington committed:
[t2430] Fixing broken/incorrect links, or disabling them
Fixing whitespace

stenington committed:
[t2430] Putting in requested projects, sans graphics and proper authors

stenington committed:
[t2430] Adding in logo, modified from larger popcornlogo.png

stenington committed:
[t2430] Size video to fit 4 column span; plus tweaks

stenington committed:
[t2430] Add js detection to page

stenington committed:
[t2430] Remove bare project button

stenington committed:
[t2430] Add projects carousel and calls to action

stenington committed:
[t2430] background and font style tweaks

mjschranz commented:
/botio preview

ghost commented:
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @mjschranz received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/585f68bb67e386f/output.txt

ghost commented:
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/585f68bb67e386f/output.txt

Total script time: 1.61 mins

Server is running on http://pr1070.butyr.org/templates/basic/
Run the Test Suite: http://pr1070.butyr.org/test/

k88hudson commented on public/bestevar.html:
```diff
@@ -0,0 +1,80 @@
+<!doctype html>
+<!-- ==========================================================================
```
Why are we adding this?

stenington replied:
Oh, bestevar.html could probably be removed. I just didn't have the heart to do away with it.

k88hudson commented on public/index.html:
```diff
@@ -1,80 +1,272 @@
-<!-- ==========================================================================
-WE MADE THIS IN THIMBLE CUZ WE CAN! thimble.webmaker.org ========================================================================= -->
+<!DOCTYPE html>
+<html lang="en-US" dir="ltr" class="no-js">
```
class="no-js" not needed

stenington replied:
There are `no-js` fallbacks for the carousel so that the page still looks reasonable without it.

k88hudson commented on public/index.html:
```diff
@@ -1,80 +1,272 @@
+<head>
+  <meta charset="utf-8">
+  <title>Mozilla Popcorn</title>
+  <link href="/vendor/video-js/video-js.min.css" rel="stylesheet">
```
Do we need these? Probably not

k88hudson commented on public/index.html:
```diff
@@ -1,80 +1,272 @@
+  <script>
+    _V_.options.flash.swf = "/vendor/videojs/video-js.swf"
+  </script>
+  <link rel="stylesheet" href="/media/css/opensans/stylesheet.css">
```
Open sans is already included in our tree

k88hudson commented on public/index.html:
```diff
@@ -1,80 +1,272 @@
+  <link href="//www.mozilla.org/tabzilla/media/css/tabzilla.css" rel="stylesheet">
+  <!--[if lt IE 9]>
+    <script src="/media/js/html5shiv.js"></script>
+  <![endif]-->
```
Is this necessary?

stenington replied:
I believe it's necessary since the page does use `section` tags, etc, but I didn't actually give it thought here. It just came over from Thimble so I left it.

k88hudson commented on public/index.html:
```diff
@@ -1,80 +1,272 @@
+</head>
+<body id="home">
+  <script type="text/javascript">
+    document.documentElement.className = document.documentElement.className.replace(/\bno-js\b/, '');
```
remove

k88hudson commented on public/index.html:
```diff
@@ -1,80 +1,272 @@
+            <div class="span10">
+              <div class="logo">
+                <a href="/en-US/">
+                  <img src="/popcornlogo-small.png" alt="Mozilla Thimble">
```
Alt text

stenington replied:
Ha... whoops. I'll change this.

k88hudson commented on public/index.html:
```diff
@@ -1,80 +1,272 @@
+              </p>
+            </div>
+            <div class="span4">
+              <video class="video-js vjs-default-skin" controls preload="none" width="344" height="232" 
```
Can we do this without videojs?

stenington replied:
I used videojs because that's what mozillapopcorn.org uses and I assumed it met some level of desired backwards-compatibility, but if straight `<video>` tags are good enough, I can switch it.

stenington replied:
@k88hudson Should I remove videojs?

k88hudson replied:
Yes, let's remove -- we'll use our own embed stuff in anything.

k88hudson commented on public/index.html:
```diff
@@ -1,80 +1,272 @@
+
+  <script src="//www.mozilla.org/media/js/libs/jquery-1.7.1.min.js"></script>
+  <script src="//www.mozilla.org/tabzilla/media/js/tabzilla.js"></script> 
+  <script src="/media/js/jcarousellite_1.0.1.min.js"></script>
```
This should be relative, i.e. `js/jcarousel....`

stenington replied:
I don't mind changing it, but why is a relative url preferable here?

k88hudson replied:
I was thinking we we're sure where this page would end up, but I guess it will always be in the root directory of popcorn.webmaker.org

k88hudson commented on public/bestevar.html:
```diff
@@ -0,0 +1,80 @@
+        background-color: white;
+        color: black;
+        padding-top: 1em;
+        font-family: "Press Start 2P";
```
Let's just use open sans

k88hudson commented on public/bestevar.html:
```diff
@@ -0,0 +1,80 @@
+      #lesson ol li {
+        padding-bottom: 20px;
+        padding-left: 10px;
+        text-align:left;
```
spacing

k88hudson commented on public/bestevar.html:
```diff
@@ -0,0 +1,80 @@
+      }
+
+      #narrative {
+        font-family: "Press Start 2P";
```
open sans

stenington committed:
[t2430] Use existing webfonts.less

* Added additional webfonts.css build to make.js
* Cleaned out /public/media/css/opensans directory, but
unfortunately for webfonts.css to work from public dir
had to copy resources/fonts to public/media as well.
* Substantially reduced sandstone.css, which was a wreck
* Changed wpm.css to use different Open Sans definitions

stenington committed:
[t2430] Changed Thimble to Popcorn in img alt, and changed obsolete css file name

stenington committed:
[t2430] Removed tons of obsolete images, moved only required sandstone/ img up a dir.

stenington committed:
[t2430] removing old index page

stenington committed:
[t2430] Remove videojs

stenington committed:
[t2430] Removing more style leftovers

stenington committed:
[t2430] Relativize links

k88hudson commented:
I'm moving this over to my fork so we can take it from here. Thanks Mike!
